src/GraphQL/Execution/DefaultExecutionStrategySelector.cs

```diff
@@ -0,0 +1,52 @@
+using GraphQLParser.AST;
+
+namespace GraphQL.Execution
+{
+    /// <inheritdoc cref="IExecutionStrategySelector"/>
+    public class DefaultExecutionStrategySelector : IExecutionStrategySelector
+    {
+        private readonly ExecutionStrategyRegistration[] _registrations;
+
+        /// <summary>
+        /// Initializes an instance that only returns the default registrations;
+        /// <see cref="ParallelExecutionStrategy"/> for <see cref="OperationType.Query"/> and
+        /// <see cref="SerialExecutionStrategy"/> for <see cref="OperationType.Mutation"/>.
+        /// </summary>
+        internal DefaultExecutionStrategySelector()
+            : this(Array.Empty<ExecutionStrategyRegistration>())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance with the specified registrations.
+        /// If no registration is specified for <see cref="OperationType.Query"/>, returns <see cref="ParallelExecutionStrategy"/>.
+        /// If no registration is specified for <see cref="OperationType.Mutation"/>, returns <see cref="SerialExecutionStrategy"/>.
+        /// </summary>
+        public DefaultExecutionStrategySelector(IEnumerable<ExecutionStrategyRegistration> registrations)
+        {
+            _registrations = registrations is ExecutionStrategyRegistration[] array
+                ? array
+                : (registrations ?? throw new ArgumentNullException(nameof(registrations))).ToArray();
```

Comment:
```suggestion
            _registrations = (registrations ?? throw new ArgumentNullException(nameof(registrations))).ToArray();
```

I don't think that optimization matters here. This will only cause extra questions. In other places (enumerable in ctor) we do not have such behavior.

Reply:
Ok.   Just FYI, MS DI uses arrays under the hood.  But sure, it's only one allocation during startup.

Reply:
Yes, I know.